app/src/main/java/com/github/factotum_sdp/factotum/ui/roadbook/RoadBookViewModel.kt

```diff
@@ -90,15 +104,53 @@ class RoadBookViewModel(_dbRef: DatabaseReference) : ViewModel() {
     }
 
     /**
-     * Edit the DestinationRecort at indec pos in the recordsList attribute
+     * Edit the DestinationRecord at index pos in the recordsList attribute
+     * If the new DestRecord computed is the same at the old one no value is posted and false is returned
      * @param pos: Int position Index at which the current DestRecord will be override
-     * @param newRec: DestinationRecord The record containing the new data
+     * @param clientID The Customer unique identifier associated to this DestinationRecord
+     * @param timeStamp The arrival time
+     * @param waitingTime The waiting time in minutes
+     * @param rate Rate as internal code notation
+     * @param actions The actions to be done on a destination
+     * @param notes The additional notes concerning a destination
+     * @return true if according the args, there is a change and the _recordList is updated, false otherwise
      */
-    fun editRecord(pos: Int, newRec: DestinationRecord) {
+    fun editRecord(pos: Int, clientID: String, timeStamp: Date?, waitingTime: Int,
+                   rate: Int, actions: List<DestinationRecord.Action>, notes: String): Boolean {
+        val currentRec = _recordsList.value!![pos]
+        var destID = currentRec.destID
+        if(currentRec.clientID != clientID) {
+            destID = computeDestID(clientID)
+        }
+        val newRec = DestinationRecord(destID, clientID, timeStamp, waitingTime, rate, actions, notes)
         val ls = arrayListOf<DestinationRecord>()
         ls.addAll(_recordsList.value as Collection<DestinationRecord>)
         ls[pos] = newRec
-        _recordsList.postValue(ls)
+        if(currentRec != newRec) {
+            _recordsList.postValue(ls)
+            return true
+        }
+        // Prefer to be explicit with a boolean value, for the front-end to know it has to refresh, or act accordingly.
+        // ! Check the case where the destID is the same but
+        return false
+    }
+
+    //Needed to update the destIDOccurrences cache
+    private fun addDemoRecords(ls: List<DestinationRecord>) {
+        val newList = arrayListOf<DestinationRecord>()
+        ls.forEach {
+            val destID = computeDestID(it.clientID)
+            newList.add(DestinationRecord(destID, it.clientID, it.timeStamp, it.waitingTime, it.rate, it.actions, it.notes))
+        }
+        _recordsList.postValue(newList)
+    }
+
+    private fun computeDestID(clientID: String): String {
```

Comment:
I don't think this code will cause any issue in our case, but if two threads run it, it has race condition in it so be careful.